packages/node-cli/src/core/sdksetup/ProxyAgent.js

```diff
@@ -0,0 +1,92 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+const http = require('http');
+const { URL } = require('url');
+const tls = require('tls');
+const net = require('net');
+
+class ProxyAgent {
+	constructor(proxyString, options) {
+		this.proxyString = proxyString;
+		this.options = options;
+		this.initProxyURL();
+	}
+
+	initProxyURL() {
+		// throws error if proxyString is invalid
+		const proxy = new URL(this.proxyString);
+
+		proxy.host = proxy.hostname || proxy.host;
+		proxy.port = +proxy.port || (proxy.protocol.toLowerCase() === 'https:' ? 443 : 80);
+		this.proxyURL = proxy;
+	}
+
+	addRequest(req, options) {
+		req.shouldKeepAlive = false;
+
+		this.createConnection(options)
+			.then((socket) => {
+				req.onSocket(socket);
+			})
+			.catch((err) => {
+				req.emit('error', err);
+			});
+	}
+
+	createConnection(options) {
+		return new Promise((resolve, reject) => {
+			const ssl = options.protocol ? options.protocol.toLowerCase() === 'https:' : false;
+			if (ssl && this.options.tunnel === true) {
```

Comment:
Can ssl be undefined here or why are you testing it?

Reply:
We only want to use the tls/ssl when donwload url is on https: . That's what we check here. I'll rename the variable name to make it more readable.